experiments/hyperparameter_search/launch_sweep.py

```diff
@@ -0,0 +1,174 @@
+            origin=np.array(args.ref_point),
+            wandb_entity=args.wandb_entity,
+            **config,
+            seed=seed,
+            group=group
+        )
+
+        # Launch the agent training
+        print(f"Worker {worker_num}: Seed {seed}. Training agent...")
+        algo.train(
+            total_timesteps=args.num_timesteps,
+            eval_env=eval_env,
+            ref_point=np.array(args.ref_point),
+            known_pareto_front=None,
+            **args.train_hyperparams,
+        )
+
+    else:
+        print(f"Worker {worker_num}: Seed {seed}. Instantiating {args.algo} on {args.env_id}")
+        env = MORecordEpisodeStatistics(mo_gym.make(args.env_id), gamma=config["gamma"])
+        eval_env = mo_gym.make(args.env_id)
+
+        algo = ALGOS[args.algo](
+            env=env,
+            wandb_entity=args.wandb_entity,
+            **config,
+            seed=seed,
+            group=group
+        )
+
+        if args.env_id in ENVS_WITH_KNOWN_PARETO_FRONT:
+            known_pareto_front = env.unwrapped.pareto_front(gamma=config["gamma"])
+        else:
+            known_pareto_front = None
+
+        # Launch the agent training
+        print(f"Worker {worker_num}: Seed {seed}. Training agent...")
+        algo.train(
+            eval_env=eval_env,
+            ref_point=np.array(args.ref_point),
+            known_pareto_front=known_pareto_front,
+            **args.train_hyperparams,
+        )
+
+        # Get the hypervolume from the wandb run
+        hypervolume = wandb.run.summary["eval/hypervolume"]
+        print(f"Worker {worker_num}: Seed {seed}. Hypervolume: {hypervolume}")
+
+        return WorkerDoneData(hypervolume=hypervolume)
+
+def main():
+    # Get the sweep id
+    sweep_run = wandb.init()
+
+    # Spin up workers before calling wandb.init()
+    # Workers will be blocked on a queue waiting to start
+    with ProcessPoolExecutor(max_workers=args.num_seeds) as executor:
+        futures = []
+        for num in range(args.num_seeds):
+            # print("Spinning up worker {}".format(num))
+            seed = seeds[num]
+            futures.append(executor.submit(train, WorkerInitData(
+                sweep_id=sweep_id,
+                seed=seed,
+                config=dict(sweep_run.config),
+                worker_num=num
+            )))
+
+        # Get results from workers
+        results = [future.result() for future in futures]
+
+    # Get the hypervolume from the results
+    metrics = [result.hypervolume for result in results]
+
+    # Compute the average hypervolume
+    average_hypervolume = sum(metrics) / len(metrics)
+    print(f"Average hypervolume of the sweep {sweep_id}: {average_hypervolume}")
+
+    # Log the average hypervolume to the sweep run
+    sweep_run.log(dict(hypervolume=average_hypervolume))
+    wandb.finish()
+
+args = parse_args()
+# print(args)
+
+# Create an array of seeds
+seeds = [random.randint(0, 1000000) for _ in range(args.num_seeds)]
```

Comment:
I would prefer to have seeds which are deterministic (for reproducibility), so just do from 0 to n-1

Reply:
I've added `--seed` argument to the launcher, which calls  `seed_everything(args.seed)`. I guess this is not really necessary to call `seed_everything()` . Probably it would have been enough to just call `random.seed(seed)` but since it's already there I've reused it.

Reply:
So the seeds within the sweeps are picked deterministically now